aws/src/main/java/org/apache/iceberg/aws/AssumeRoleCredentialsFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws;
+
+import java.util.Map;
+import java.util.UUID;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.PropertyUtil;
+import software.amazon.awssdk.awscore.client.builder.AwsClientBuilder;
+import software.amazon.awssdk.services.sts.StsClientBuilder;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+
+/**
+ * Factory to use STS assume role credential to access cross-account cross-region AWS resources.
+ */
+public class AssumeRoleCredentialsFactory implements AwsClientCredentialsFactory {
+
+  private String roleArn;
+  private String externalId;
+  private int timeout;
+
+  @Override
+  public void configure(AwsClientBuilder clientBuilder) {
+    // STS will use default credential in the environment as the root credential to assume the role
+    if (!(clientBuilder instanceof StsClientBuilder)) {
```

Comment:
you need to check if the caller has called init before calling configure.

Reply:
initialize is called from engines like Spark that passes user configurations, and is guaranteed to be called as the fist method during dynamic loading.